packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -711,4 +789,62 @@ - (void)setMixWithOthers:(FLTMixWithOthersMessage *)input
   }
 }
 
+- (nullable NSNumber *)isPictureInPictureSupported:
+    (FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  return @([AVPictureInPictureController isPictureInPictureSupported] &&
+           [self doesInfoPlistSupportPictureInPicture]);
+}
+
+- (void)setAutomaticallyStartsPictureInPicture:
+            (FLTAutomaticallyStartsPictureInPictureMessage *)input
+                                         error:(FlutterError **)error {
+  FLTVideoPlayer *player = self.playersByTextureId[input.textureId];
+  [player
+      setAutomaticallyStartsPictureInPicture:input
+                                                 .enableStartPictureInPictureAutomaticallyFromInline
+                                                 .boolValue];
+}
+
+- (void)setPictureInPictureOverlaySettings:(FLTSetPictureInPictureOverlaySettingsMessage *)input
+                                     error:(FlutterError **)error {
+  FLTVideoPlayer *player = self.playersByTextureId[input.textureId];
+  [player setPictureInPictureOverlaySettings:CGRectMake(input.settings.left.floatValue,
+                                                        input.settings.top.floatValue,
+                                                        input.settings.width.floatValue,
+                                                        input.settings.height.floatValue)];
+}
+
+- (BOOL)doesInfoPlistSupportPictureInPicture {
+  NSArray *backgroundModes = [NSBundle.mainBundle objectForInfoDictionaryKey:@"UIBackgroundModes"];
+  return ![backgroundModes isKindOfClass:[NSArray class]] ||
+         ![backgroundModes containsObject:@"audio"];
+}
+
+- (void)startPictureInPicture:(FLTStartPictureInPictureMessage *)input
+                        error:(FlutterError **)error {
+  if (![self doesInfoPlistSupportPictureInPicture]) {
+    *error = [FlutterError
+        errorWithCode:@"video_player"
+              message:@"Failed to start picture-in-picture because UIBackgroundModes: audio, "
+                      @"AirPlay, picture-in-picture is not enabled in Info.plist"
```

Comment:
Here and below, what is the "AirPlay, picture-in-picture" part? Only `UIBackgroundModes: audio` is being changed in the method above.

Reply:
In the UI it is called `Audio, Airplay and Picture In Picture` That is why I wanted to use the same references. How would you structure this message?

![image](https://github.com/flutter/packages/assets/21172855/b327b34c-fc6b-4337-9f48-2417d5cd5cee)

Reply:
I would suggest checking what the Info.plist key is actually called (rather than the current UI display of that key, which could change at any time) and referencing that.

Reply:
Fair point! I will fix that.

Reply:
`audio` is the key that should be added to the `UIBackgroundModes`. So I updated the text